### PR TITLE
[NUI] Add Dashed/Double type to Text Underline

### PR DESCRIPTION
**Pull request by wonrst**
Added Solid/DashWidth/DashGap types to the Underline struct.
If not provided, default is Solid type.
No effect on legacy underline behavior.

- Added Enum UnderlineType (Solid, Dashed, Double)

- Added new properties to Underline struct
  - Type : The type of the underline (the default type is Solid)
  - DashWidth : The width of the dashes of the dashed underline
  - DashGap : The gap between the dashes of the dashed underline

dali-toolkit:
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/257630/

Signed-off-by: Bowon Ryu <bowon.ryu@samsung.com>

### CODE REVIEW

**Review by everLEEst — APPROVED** [review]
Looks good to me.

File: src/Tizen.NUI/src/public/BaseComponents/TextConstants.cs [hunk 1]
```diff
@@ -135,6 +140,18 @@ public struct Underline
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public float? Height { get; set; }
+
+        /// <summary>
+        /// The width of the dashes of the dashed underline (if null, the default value is 2.0f). <br />
```
Comment (Seoyeon2Kim) [inline src/Tizen.NUI/src/public/BaseComponents/TextConstants.cs:145]: One question :
In DALi, the default value is 1 pixel. However, in NUI here, the default value is 2.0f. Is there any reason that they have different values?
-> "If not provided then the default width is used (1 pixel)."
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/257630/38/dali-toolkit/devel-api/controls/text-controls/text-style-properties-devel.h

Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextConstants.cs:145]: Because DALi's description is incorrect..
I will fix the description of DALi.

Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextConstants.cs:145]: Plese refer this.
```
  mUnderlineType(Text::Underline::SOLID),
  mDashedUnderlineWidth(2.0f),
  mDashedUnderlineGap(1.0f),
```
https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/257630/38/dali-toolkit/internal/text/visual-model-impl.cpp

Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/TextConstants.cs:145]: Oh I see :)
Thanks for updating DALi description fast.

**Review by Seoyeon2Kim — APPROVED** [review]
LGTM! :)

**Review by jaehyun0cho — APPROVED** [review]
LGTM

**Comment by Seoyeon2Kim** [comment]
@wonrst , Can we merge this PR now?
Could you check it one more time?

**Pull request merged**